design-proposals/klip-55-tablespaces.md

```diff
@@ -1,3 +1,210 @@
 # KLIP 55 - Tablespaces
 
-Just a placeholder. :-)
+**Author**: @MichaelDrogalis | 
+**Release Target**: TBD | 
+**Status**: In Discussion | 
+**Discussion**: _to be added_
+
+Tablespaces help you use knowledge of the usage pattern of a set of ksqlDB objects to optimize performance.
+
+## Motivation and background
+
+Real-world SQL programs make use of many kinds of objects. Sometimes,
+a subset of those objects have a specific usage pattern. If you could tune for
+that usage pattern on a granular level, ksqlDB could treat those objects more efficiently.
+
+Tablespaces are a construct for defining groups of configuration that can be applied
+to one or more database objects. They exist in
+[Postgres](https://www.postgresql.org/docs/10/manage-ag-tablespaces.html),
+[MySQL](https://dev.mysql.com/doc/refman/8.0/en/general-tablespaces.html),
+[Oracle](https://docs.oracle.com/cd/A57673_01/DOC/server/doc/SCN73/ch4.htm), and so on.
+In the context on ksqlDB, a tablespace would allow you to override default server
+configuration for one or more objects.
+
+This is particularly important given the impact on how ksqlDB's runtime is being rearchitected.
+When multiple persistent queries begin to share a runtime, it will become more challenging to
+vary configuration parameters across them. Tablespaces give you a logical boundary to hint to
+ksqlDB where it must respect varying parameters.
+
+Tablespaces don't enforce namespacing or scoping: two objects in different tablespaces
+can be used with one another.
+
+## What is in scope
+
+- The grammar to support tablespace commands
+- The behavior of those commands
+- Backward compatibility
+
+## What is not in scope
+
+- How it's implemented
+
+## Value/Return
+
+- Improved program performance
+- Isolation of functional parameters to a subset of objects
+
+## Public APIs
+
+**Using the default tablespace**
+
+Creating an object works like normal—it's placed into a default tablespace so
+that you never need to use this feature unless you want to.
+
+```sql
+CREATE stream object WITH (
+    ...
+);
+```
+
+The default tablespace is `ksql_default`.
```

Comment:
not sure the additional 'ksql_' prefix adds much value here?

Reply:
Open to anything here, just copied what Postgres does. `default` seems just as good.

Reply:
I've seen other Dbs use `default` too.